### PR TITLE
Position title after legend width calculated

### PR DESCRIPTION
**Pull request by 28raining**
To continue fixing https://github.com/plotly/plotly.js/issues/6668#issuecomment-1709202910

The original method worked because the legend-draw function is called multiple times. I don't know why, it seems like a mistake. On the 2nd call, the legend has been drawn and `legendObj._width` has been calculated.

Sometimes the function is only called once. I was able to reproduce by changing the height of one of the mocks. This mean legend horizontal align didn't work.

I've updated the code to re-draw the legend title after `computeLegendDimensions`
- the first draw allows the title height to be accounted for in `computeLegendDimensions`
- the second draw just has purpose of horizontal aligning the title

### CODE REVIEW

**Comment by 28raining** [comment]
I've bundled a couple of other issue fixes in here

https://github.com/plotly/plotly.js/issues/6661
- single-point histograms have an existing special workaround to calculate the bin width. But that shouldn't run if the user has defined the bin widths manually, or assigned a bin-group
- The single-point box is wider than the multi-point box, you can see in my new mock. That's because May is longer than June. Should bars be as wide as the month? Or should all bars be the same width. I think it doesn't look too bad and is a rare corner case, so I'm happy to leave it.

https://github.com/plotly/plotly.js/issues/6685
- I did file a PR on color-parse, and they recently released v2.0.0. Which I assume includes the `.toLowerCase()` fix

**Comment by 28raining** [comment]
I'm stumped, the tests pass on my personal dockers, and regenerating the images isn't helping

File: package.json [hunk 1]
```diff
@@ -79,7 +79,7 @@
     "canvas-fit": "^1.5.0",
     "color-alpha": "1.0.4",
     "color-normalize": "1.5.0",
-    "color-parse": "1.3.8",
+    "color-parse": "2.0.0",
```
Comment (archmoj) [inline package.json:82]: Please revert this dependency change. Upgrades like this could be proposed in a separate PR.

File: test/image/mocks/legend_horizontal_one_row.json [hunk 2]
```diff
@@ -20,6 +20,7 @@
         }
     ],
     "layout": {
+        "height": 300,
```
[the remark this answers is not in the document]
Comment (28raining) [reply test/image/mocks/legend_horizontal_one_row.json:23]: This was how I recreated the issue in the ticket.

File: test/image/mocks/zz_histogram_overlay-1point-date.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        }
+      },
+      "xbins": {
+        "size": "M1",
+        "start": "2023-05-01",
+        "end": "2023-07-01"
+      }
+    }
+  ],
+  "layout": {
```
[the remark this answers is not in the document]
Comment (28raining) [reply test/image/mocks/zz_histogram_overlay-1point-date.json:40]: This is to prove that the fix a different ticket is working; https://github.com/plotly/plotly.js/issues/6661

**Comment by archmoj** [comment]
Please write `draftlog/6725_fix.md` file mentioning two different fixes and this should be good to go.
Thank you! :medal_sports: :trophy:
